src/main/java/seedu/address/model/job/Job.java

```diff
@@ -25,6 +27,8 @@ public class Job {
      */
     public Job(Name name, JobCompany company, JobSalary salary,
             JobDescription description, Set<Tag> requirements) {
+        requireAllNonNull(name, company, salary, description);
+        requireAllNonNull(requirements);
```

Comment:
Is it possible to put requirements inside the first requireAllNonNull()?

Reply:
Ah I see, there might be some big implications, although I am not sure how a `null` value would enter the collection.